src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -507,7 +509,10 @@ def setUp(self):
         self.fault_injection = FaultInjection()
         self.fault_injection.start(self.params.get("fault_list", '/run/faults/*'), self.test_dir)
 
-        self.context = DaosContext(self.prefix + '/lib64/')
+        self.context = DaosContext(os.path.join(self.prefix,
+                                                'lib', 'x86_64-linux-gnu'
+                                                if detect().name.lower() == 'ubuntu'
+                                                else 'lib64'))
```

Comment:
The if statement only acts on the last argument:
```
>>> os.path.join('/', 'lib', 'x86_64-linux-gnu' if False else 'lib64')
'/lib/lib64'
```
I suspect you want this behavior:
```
>>> os.path.join('/', os.path.join('lib', 'x86_64-linux-gnu') if False else 'lib64')
'/lib64'
>>> os.path.join('/', os.path.join('lib', 'x86_64-linux-gnu') if True else 'lib64')
'/lib/x86_64-linux-gnu'
```
```suggestion
        lib_dir = os.path.join('lib', 'x86_64-linux-gnu') if detect().name.lower() == 'ubuntu' else 'lib64'
        self.context = DaosContext(os.path.join(self.prefix, lib_dir))
```

Reply:
Except that the two path choices are `/lib/x86_64-linux-gnu` and `/usr/lib64`.  I'll take another look at this code and see if it is currently doing as I am hoping.  Thanks for pointing out that there may be issues here though.